.github/workflows/build-windows-vhd.yml

```diff
@@ -1,104 +1,159 @@
-name: Build or Test on Hyper-V Runner
+name: Build and Upload Windows VHD
 
 on:
   push:
     branches:
       - main
+      - user/vhd_pipeline_automation
+  workflow_dispatch:
+    inputs:
+      kubernetes_version:
+        description: 'Kubernetes version (optional, leave blank for latest)'
+        required: false
+        default: ''
+      windows_version:
+        description: 'Windows version (optional, leave blank for latest)'
+        required: false
+        default: ''
 
 jobs:
-  build-vhd:
-    name: Run on Hyper-V Self-Hosted Runner
+  build-upload-vhd:
+    name: Build and Upload VHD
     runs-on:
       - self-hosted
       - windows
       - hyperv
 
+    env:
+      AZURE_STORAGE_ACCOUNT: ${{ secrets.AZURE_STORAGE_ACCOUNT }}
+      AZURE_STORAGE_KEY:     ${{ secrets.AZURE_STORAGE_KEY }}
+      AZURE_CONTAINER_NAME:  ${{ secrets.AZURE_STORAGE_CONTAINER }}
+
     steps:
       - name: Checkout repository
         uses: actions/checkout@v4
 
       - name: Display runner info
+        shell: pwsh
         run: |
-          echo "Runner OS: $RUNNER_OS"
-          echo "Runner Labels: $RUNNER_LABELS"
-          echo "PowerShell version: $($PSVersionTable.PSVersion)"
+          echo "Runner OS:        $env:RUNNER_OS"
+          Write-Host "Runner OS:        $env:RUNNER_OS"
+          echo "Runner Labels:    $env:RUNNER_LABELS"
+          Write-Host "Runner Labels:    $env:RUNNER_LABELS"
+          echo "PowerShell Ed.:   $($PSVersionTable.PSEdition) $($PSVersionTable.PSVersion)"
+          Write-Host "PowerShell Ed.:   $($PSVersionTable.PSEdition) $($PSVersionTable.PSVersion)"

```

Comment:
Why the duplicate print statements?

Reply:
Got ride of the echo statements since the Write-Host should suffice!